crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -379,20 +384,27 @@ struct LinePreprocessor<'a> {
     /// Maximum number of consecutive blank lines between the current line and the previous non-comment logical line.
     /// One of its main uses is to allow a comment to directly precede a class/function definition.
     max_preceding_blank_lines: BlankLines,
+    /// The cell offsets of the notebook (if running on a notebook).
+    cell_offsets: Option<Peekable<Iter<'a, TextSize>>>,
+    /// If running on a notebook, whether the line is the first logical line (or a comment preceding it) of its cell.
+    is_beginning_of_cell: bool,
 }
 
 impl<'a> LinePreprocessor<'a> {
     fn new(
         tokens: &'a [LexResult],
         locator: &'a Locator,
         indent_width: IndentWidth,
+        cell_offsets: Option<&'a CellOffsets>,
     ) -> LinePreprocessor<'a> {
         LinePreprocessor {
             tokens: tokens.iter(),
             locator,
             line_start: TextSize::new(0),
             max_preceding_blank_lines: BlankLines::Zero,
             indent_width,
+            is_beginning_of_cell: cell_offsets.is_some(),
+            cell_offsets: cell_offsets.map(|cell_offsets| cell_offsets[1..].iter().peekable()),
```

Comment:
I wonder if this panics for an empty file (that contains no new lines). It might be safer to use `cell_offsets.get(1..).unwrap_or_default().iter().peekable()`

Reply:
Done in [05a7473](https://github.com/astral-sh/ruff/pull/10291/commits/05a747381f5efb2509fd4c1c2512dd55c604fb09).